### PR TITLE
Solace AMQP PubSub Component

### PR DESCRIPTION
**Pull request by TKTheTechie**
# Description

[Solace](https://solace.com/products/event-broker/software/) is a multi-protocol event broker that supports AMQP, MQTT, REST, and a native Solace Protocol. I've created a pubsub component with the go-AMQP 1.0 library.

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: #2382 

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [x] Created/updated tests
* [x] Extended the documentation _

### CODE REVIEW

**Review by TKTheTechie — COMMENTED** [review]
Signed-off-by: TKTheTechie <thomas.kunnumpurath@solace.com>

File: tests/config/pubsub/tests.yml
Comment (berndverst) [inline tests/config/pubsub/tests.yml:15]: why change the type of quote? doublequotes was just fine here

**Review by berndverst — CHANGES_REQUESTED** [review]
Please also edit `.github/workflows/conformance.yml` in the same way as was done for `kubemq` pubsub. You will have to add your pubsub component to the test matrix, and also ensure an appropriate docker image is started -- for that you may need to add a docker-compose file to `.github/infrastructure`.

**Comment by berndverst** [comment]
Can you invite me with write access to your fork so I can merge the master branch into your PR? Our PRs quickly get out of date and coordinating with PR authors is difficult, merging your PR can take a long time as a result.

Additionally I suspect that the linter will fail on some things (e.g. no space after `//` comments). Please make sure to run `make lint` and address any issues that come up.

**Comment by TKTheTechie** [comment]
> Can you invite me with write access to your fork so I can merge the master branch into your PR? Our PRs quickly get out of date and coordinating with PR authors is difficult, merging your PR can take a long time as a result.
> 
> Additionally I suspect that the linter will fail on some things (e.g. no space after `//` comments). Please make sure to run `make lint` and address any issues that come up.

Thanks. I think I fixed/added everything and have given you write access on my fork.

File: pubsub/solace/amqp/amqp.go
Comment (DeepanshuA) [inline pubsub/solace/amqp/amqp.go:131]: I think as per tracking issue https://github.com/dapr/components-contrib/issues/1797, we need to disable built-in re-tries for publishing as well. So, may be in new components, it may not be a recommendable practice to include this.
Thoughts @berndverst @ItalyPaleAle?

Comment (berndverst) [reply pubsub/solace/amqp/amqp.go:131]: We can keep this for now to unblock this. There will be more components which need updating and  the strategy for this consistently isn't totally clear yet @DeepanshuA.

Comment (DeepanshuA) [reply pubsub/solace/amqp/amqp.go:131]: Sure, thanks for the clarification @berndverst

**Review by berndverst — CHANGES_REQUESTED** [review]
There are a bunch of remaining linter failures and compilation fails -- it seems the amqp package methods you are using don't exist in our version.

**Comment by berndverst** [comment]
```
pubsub/solace/amqp/amqp.go:254:63  typecheck  ConnOption not declared by package amqp
pubsub/solace/amqp/amqp.go:52:9    typecheck  cannot use &(amqpPubSub literal) (value of type *amqpPubSub) as pubsub.PubSub value in return statement: *amqpPubSub does not implement pubsub.PubSub (wrong type for method Publish)
                                                      have Publish(req *pubsub.PublishRequest) error
                                                      want Publish(ctx context.Context, req *pubsub.PublishRequest) error
pubsub/solace/amqp/amqp.go:119:8   typecheck  LinkTargetAddress not declared by package amqp
pubsub/solace/amqp/amqp.go:151:8   typecheck  LinkSourceAddress not declared by package amqp
pubsub/solace/amqp/amqp.go:217:53  typecheck  cannot use ... in call to non-variadic amqp.Dial
pubsub/solace/amqp/amqp.go:223:36  typecheck  not enough arguments in call to client.NewSession
                                           have ()
                                           want (context.Context, *"github.com/Azure/go-amqp".SessionOptions)
pubsub/solace/amqp/amqp.go:255:18  typecheck  ConnOption not declared by package amqp
pubsub/solace/amqp/amqp.go:262:29  typecheck  ConnSASLAnonymous not declared by package amqp
pubsub/solace/amqp/amqp.go:264:29  typecheck  ConnSASLPlain not declared by package amqp
pubsub/solace/amqp/amqp.go:267:28  typecheck  ConnSASLPlain not declared by package amqp
```

**Comment by berndverst** [comment]
@TKTheTechie we are using azure/go-amqp v0.18.0 which had several breaking changes. More info here:
https://github.com/Azure/go-amqp/blob/main/CHANGELOG.md#0180-2022-12-06

You will need to update your PR to deal with these breaking changes. Sorry about that.

**Comment by TKTheTechie** [comment]
> @TKTheTechie we are using azure/go-amqp v0.18.0 which had several breaking changes. More info here: https://github.com/Azure/go-amqp/blob/main/CHANGELOG.md#0180-2022-12-06
> 
> You will need to update your PR to deal with these breaking changes. Sorry about that.

Yes, apologies for not catching that. Its been fixed now and tests are passing.

**Review by berndverst — APPROVED** [review]

**Pull request merged**
